javascript/packages/fury/lib/codeGen.ts

```diff
@@ -87,6 +88,19 @@ function typeHandlerDeclaration(fury: Fury) {
             return genTagDeclaration(Cast<ObjectTypeDescription>(description).options.tag);
         }
         if (description.type === InternalSerializerType.ARRAY) {
+            const tupleOptions = Cast<TupleTypeDescription>(description).options;
+            if (tupleOptions && tupleOptions.isTuple) {
+                const names = [] as string[];
+                Cast<TupleTypeDescription>(description).options.inner.forEach(v => {
+                    names.push(genDeclaration(v));
+                })
+
+                const name = `tuple_${names.join('_')}`;
```

Comment:
after some investigation, it seem the `addDeclar` function can avoid declare same variable twice.

and if two tuple has the same name, that means it can serialize the same data.

and also I have add an code gen snapshot to check the generated code.